hedera-node/src/main/java/com/hedera/services/state/logic/RecordStreaming.java

```diff
@@ -75,9 +97,46 @@ public void run() {
 	}
 
 	public void stream(final RecordStreamObject rso) {
+		if (isInNewBlock(rso)) {
+			Hash computedHash;
+			try {
+				computedHash = curRunningHashLeaf.getLatestBlockHash();
+			} catch (final InterruptedException e) {
+				log.error("Error in computing hash for block #{}", curNetworkCtx.getBlockNo());
+				Thread.currentThread().interrupt();
+				return;
+			}
+
+			curNetworkCtx.finishCurrentBlock(computedHash);
+			curNetworkCtx.startNewBlock(rso.getTimestamp());
+			log.info("Beginning block #{} @ {}", curNetworkCtx.getBlockNo(), rso.getTimestamp());
+		}
+
 		runningHashUpdate.accept(rso.getRunningHash());
+		curNetworkCtx.setLatestConsTimeOfCurrentBlock(rso.getTimestamp());
 		while (!nonBlockingHandoff.offer(rso)) {
 			/* Cannot proceed until we have handed off the record. */
 		}
 	}
+
+	public boolean isInNewBlock(final RecordStreamObject rso) {
+		final var firstBlockTime = curNetworkCtx.getFirstConsTimeOfCurrentBlock();
+		final var latestBlockTime = curNetworkCtx.getLatestConsTimeOfCurrentBlock();
+		final var consTime = rso.getTimestamp();
+		boolean result;
+		if (firstBlockTime == null) {
+			result = true;
+		} else {
+			final Duration duration = Duration.between(latestBlockTime, consTime);
+			result = getPeriod(consTime, BLOCK_PERIOD_MS) != getPeriod(firstBlockTime, BLOCK_PERIOD_MS) && duration.toNanos() >= MIN_TRANS_TIMESTAMP_INCR_NANOS;
+			if(result) {
+				log.info("consTime is {}", consTime);
+				log.info("firstBlockTime is {}", firstBlockTime);
```

Comment:
I don't think we really want to log this at "info" level logging. Every 2 seconds this will be in the log. I think this should be "debug" level logging.